task/fbc-target-index-pruning-check/0.1/fbc-target-index-pruning-check.yaml

```diff
@@ -0,0 +1,230 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "konflux"
+  name: fbc-target-index-pruning-check
+spec:
+  description: >-
+    Ensures file-based catalog (FBC) components do not remove versions of operators already added to a released catalog.
+  params:
+    - name: IMAGE_URL
+      description: Fully qualified image name.
+    - name: IMAGE_DIGEST
+      description: Image digest.
+    - name: TARGET_INDEX
+      description: Image name of target index, minus tag.
+      default: registry.redhat.io/redhat/redhat-operator-index
+    - name: OCP_VERSION
```

Comment:
I assume this task will be run after `build-image-index` in the pipeline. May I ask from where are we getting the OCP_VERSION param? I thought you will have to extract it from the fragment inside the task. Am I missing something?

Reply:
I had previously suggested to not redo logic that we were doing in the `validate-fbc` task (including the catalog rendering). It might be intended for this result to come from _that_ task. 

After the work that I started doing in https://github.com/konflux-ci/konflux-test/pull/356, my previous comments may have been over-aggressive as the primary motivation was to not duplicate scripts. We can achieve this same result of reducing duplication by instead leveraging the utility scripts.

Reply:
`validate-fbc` isn't emitting OCP_VERSION either : https://github.com/konflux-ci/build-definitions/blob/ec9936c4226e3c6e57f57a1c0e5368d4b433c925/task/validate-fbc/0.1/validate-fbc.yaml#L26 

Also, OCP_VERSION seems to be a required param. @nmars Are we confident this will work in the pipeline?

Reply:
Ah, I see you're adding it. Please ignore my previous comment